source/common/router/router.h

```diff
@@ -589,15 +593,20 @@ class Filter : Logger::Loggable<Logger::Id::router>,
 
   // Keep small members (bools and enums) at the end of class, to reduce alignment overhead.
   uint32_t retry_shadow_buffer_limit_{std::numeric_limits<uint32_t>::max()};
+  uint32_t retry_connection_failure_buffer_limit_{std::numeric_limits<uint32_t>::max()};
   uint32_t attempt_count_{1};
   uint32_t pending_retries_{0};
+  uint32_t original_buffer_limit_;
   Http::Code timeout_response_code_ = Http::Code::GatewayTimeout;
   FilterUtility::HedgingParams hedging_params_;
   bool grpc_request_ : 1;
   bool exclude_http_code_stats_ : 1;
   bool downstream_response_started_ : 1;
   bool downstream_end_stream_ : 1;
   bool is_retry_ : 1;
+  bool wait_for_connect_ : 1;
```

Comment:
I know we haven't been great about commenting this file but plase comment all new fields.

Reply:
done